### DISCUSSION TITLE
How to get block wise context for the below image

### DISCUSSION THREAD

**Discussion by nattu22** [pymupdf/PyMuPDF#2376] [Q&A]
I trying to extract a text based on blocks. Currently, I am facing this issue to split this block in to two.

![image](https://user-images.githubusercontent.com/53460886/234933742-e7aef158-0454-4317-8d7a-ddc2ba9e2059.png)
1st block in this image
![image](https://user-images.githubusercontent.com/53460886/234933870-1f68528d-1065-47fb-9261-86c19fabd6e9.png)
bottom text needs to be in 2 blocks

Eg. For words, we can specify the spaces between words and line via x_tolerence and y_tolerence.

Is there any options available in fitz. if yes pls suggest something.

**Comment by JorjMcKie** [comment]
this is a d"Discussions" item, so will be transferred.

**Reply by JorjMcKie** [reply]
Establishing the hierarchy in `TextPage` objects "block -> line -> span -> char" fully happens inside MuPDF code and cannot be influenced.
These MuPDF heuristics are quite a complex algorithm which takes a lot of parameters into account ... in an overall effort to deliver good / acceptable results in most cases.

Obviously, this may not always be ideal. So it is left to your wits to find a more adequate separation into text components in situations specific to your data.
To do this, you have all required information - down to single characters if needed. There are the extraction variants "words", "blocks", "dict" and "rawdict" to choose from.

Sorry, but I see no basis for a meaningful enhancement.
